### ISSUE TITLE
Feature suggestion: `layer_static_mapbox()` function wrapping `ggspatial::layer_spatial()`

### ISSUE THREAD

**Issue by elipousson** [walkerke/mapboxapi#17]
Thanks for a great package!

I recently ran into difficulty trying to use the `get_static_tiles()` function because I wasn't sure how to set an appropriate zoom level when returning tiles for areas of varied sizes. I had previously used the [snapbox package](https://github.com/anthonynorth/snapbox) and appreciated how the `scale_ratio` parameter from the `layer_snapbox()` function really simplified the process of getting a consistent look.

I was trying to figure out how the two packages handled this differently and realized it would be relatively straightforward to adapt a few key functions from snapbox and create an equivalent `layer_static_mapbox()` function for mapboxapi. You can [see all the changes in this commit](https://github.com/elipousson/mapboxapi/commit/262b58eeefad8f2d4a09c1fe91d14b17ec2bfdb4) but I figured an explanation would be helpful before asking if you were interested in a pull request. Here are the changes:

- I moved the code dealing with the location and buffer_dist parameters from get_static_tiles() into a new separate helper function (location_to_bbox)
- I added the location and buffer_dist parameters to static_mapbox() and dropped the bbox parameter (since location can be a bbox). 
- I set the default width and height to NULL and adapted the following code from the snapbox package which determines the width and height based on the aspect ratio of the bounding box and the scale parameter (renamed from scale_ratio in the snapbox package)

```
if (all(is.null(c(width, height)))) {
  asp <- as.numeric(abs(bbox$xmax - bbox$xmin) / abs(bbox$ymax - bbox$ymin))
  max_size <- min(1280, round(1280 * scale))
  width <- min(max_size, round(max_size * asp))
  height <- min(max_size, round(max_size / asp))
}
``` 

- I added a logical parameter (image) to static_mapbox() so it could optionally return the request rather than an image magick object.
- Finally, I created the layer_static_mapbox() function which calls static_mapbox (dropping the lat, lon, and zoom parameters and setting image to FALSE) and (after using raster to set the extent and projection) passes the image on to ggspatial::layer_spatial().

If this feature suggestion is welcome, it may be appropriate to credit Anthony North and Miles McBain since portions of the code are adapted from their generously MIT-licensed package. Let me know what you think!

**Comment by walkerke** [comment]
Thanks @elipousson - this is a really interesting adaptation.  I'll review when I get the chance!  I ended up adopting the raster tiles API for basemaps rather than the static images API because it made a bit more sense to me and __snapbox__ already accomplished it.  That said, however - __snapbox__ has not been updated in two years and it isn't on CRAN, so this may make sense as a feature.  

I'll review and ping you here if I'd like a PR, which I anticipate I will!

**Comment by elipousson** [comment]
Some trial and error this morning makes me think that `layer_static_mapbox()` is *not* assigning the CRS the way I had expected. I should have more time later this week or over the weekend to try to figure it out and fix it (unless the issue is obvious when you have a chance to review). I also added support for bbox class objects to `location_to_bbox()` as well as bbox character strings.

**Comment by elipousson** [comment]
I think the issue was just that I didn't set the CRS for the location bbox to the appropriate coordinate reference system. I think it works!

**Comment by elipousson** [comment]
I just discovered what I believe is an outstanding issue with the CRS after attempting to use layer_static_mapbox in combination with some OSM building footprint data. The API behind osmdata is error-prone so I wasn't able to create a reprex but I compared the results from mapboxapi and snapbox and you can see the difference below. I should have time over the weekend or next week to chase down this bug.

Here is the version from mapboxapi::layer_static_mapbox:

<img width="500" alt="Screen Shot 2022-04-07 at 9 22 15 AM" src="https://user-images.githubusercontent.com/931136/162221852-9aec6928-d3cd-4647-8c77-075fdae5c164.png">

And here is the version from snapbox::layer_mapbox:

<img width="500" alt="Screen Shot 2022-04-07 at 10 24 42 AM" src="https://user-images.githubusercontent.com/931136/162222606-2d8d5463-0f64-4485-abc8-827aacb959de.png">

**Comment by elipousson** [comment]
Thankfully, this turned out to be an easier fix than I had expected. I had set it up to use the lon/lat bounding box to set the width/height but the extent was based on a Mercator projected bounding box. I reworked it so the width/height is set based on the aspect ratio of the projected bounding box and it seems like it is working as expected.

Hope it is OK but I also implemented a style_url parameter to allow easier copy/pasting from Mapbox Studio into R without needing to separate the username and style_id. If you don't want it in the pull request, just let me know and I'll reverse those changes.
